src/exponential-backoff.js

```diff
@@ -16,25 +16,47 @@
 
 
 /**
- * @param {opt_base} opt_base Exponential base. Defaults to 2.
- * @return {function(function())} Function that when invoked will
+ * @param {number=} opt_base Exponential base. Defaults to 2.
+ * @return {function(function(): number)} Function that when invoked will
  *     call the passed in function. On every invocation the next
  *     invocation of the passed in function will be exponentially
- *     later.
+ *     later. Returned function returns timeout id.
  */
 export function exponentialBackoff(opt_base) {
-  let count = 0;
+  const getTimeout = exponentialBackoffClock(opt_base);
   return work => {
+    return setTimeout(work, getTimeout());
+  };
+}
+
+/**
+ * @param {number=} opt_base Exponential base. Defaults to 2.
+ * @return {function(): number} Function that when invoked will return
+ *    a number that exponentially grows per invocation.
+ */
+export function exponentialBackoffClock(opt_base) {
+  let count = 0;
+  return () => {
     let wait = Math.pow(opt_base || 2, count++);
-    // Add jitter to avoid the thundering herd. This can e.g. happen when
-    // we poll a backend and it fails for everyone at the same time.
-    // We wait up to 30% longer or shorter than the time otherwise
-    // given for this cycle.
-    let jitter = wait * .3 * Math.random();
-    if (Math.random() > .5) {
-      jitter *= -1;
-    }
-    wait += jitter;
-    setTimeout(work, Math.round(wait * 1000));
+    wait += getJitter(wait);
+    return wait * 1000;
   };
 }
+
+/**
+ * Add jitter to avoid the thundering herd. This can e.g. happen when
+ * we poll a backend and it fails for everyone at the same time.
+ * We add up to 30% (default) longer or shorter than the given time.
+ *
+ * @param {number} wait the amount if base milliseconds
+ * @param {number=} opt_perc the min/max percentage to add or sutract
+ * @return {number}
+ */
+export function getJitter(wait, opt_perc) {
```

Comment:
This seems like a super small and subjective thing to export as a function. Are you sure you need it?